File: src/main/java/no/nav/veilarboppfolging/domain/Oppfolging.java
package no.nav.veilarboppfolging.domain;

import no.nav.veilarboppfolging.repository.entity.KvpPeriodeEntity;
import no.nav.veilarboppfolging.repository.entity.OppfolgingsperiodeEntity;
import no.nav.veilarboppfolging.repository.entity.*;

import java.util.List;

import static java.util.Collections.emptyList;

public class Oppfolging {
    private String aktorId;
    private String veilederId;
    private boolean underOppfolging;
    private ManuellStatusEntity gjeldendeManuellStatus;
    private MaalEntity gjeldendeMal;
    private List<OppfolgingsperiodeEntity> oppfolgingsperioder = emptyList();
    private KvpPeriodeEntity gjeldendeKvp;
}

File: src/main/kotlin/no/nav/veilarboppfolging/domain/Oppfolging.kt
package no.nav.veilarboppfolging.domain

import no.nav.veilarboppfolging.repository.entity.KvpPeriodeEntity
import no.nav.veilarboppfolging.repository.entity.MaalEntity
import no.nav.veilarboppfolging.repository.entity.ManuellStatusEntity
import no.nav.veilarboppfolging.repository.entity.OppfolgingsperiodeEntity

data class Oppfolging(
    val aktorId: String,
    val veilederId: String?,
    val underOppfolging: Boolean,
    val gjeldendeManuellStatus: ManuellStatusEntity?,
    val gjeldendeMal: MaalEntity?,
    val oppfolgingsperioder: List<OppfolgingsperiodeEntity>,
    val gjeldendeKvp: KvpPeriodeEntity?,
)
